doc/smart-switch/PA-Validation/SmartSwitchPAValidationOffload.md

```diff
@@ -34,59 +36,114 @@ This document provides general information about offloading PA validation featur
 
-## 2.2 APPL_STATE_DB
-Based on the DPU's capability and when NPU offload is required, DashOrchagent writes all DASH_PA_VALIDATION table entries to APPL_STATE_DB. These APPL_STATE_DB entries are accessed by NPU SWSS to offload the feature to NPU. For more information of NPU and DPU database architecture, please refer to https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/smart-switch-database-architecture/smart-switch-database-design.md
+<img src="images/DashOffloadManagerWithConsumer.svg">
 
+The Dash Offload Manager is disabled by default and only enabled for specific platforms that require its functionality.
 
-## 2.3 Orchagent Changes
+### 2.1.1 SONiC GNMI server changes
+A new parameter `zmq_dpu_proxy_address_base` is added to the telemetry.go to enable the GNMI Server -> DASH Offload Manager ZMQ connection. If specified, the GNMI server will use `zmq_dpu_proxy_address_base + dpu_index:zmq_port` as a destination for DPU ZMQ connection. For example, on the system with 2 DPUs and base address 127.0.0.10, the DASH Offload Manager's ZMQ server runs on 127.0.0.10:8100 for DPU0 and on 127.0.0.11:8100 for DPU1.
 
-New orchagent(DashOffloadOrch) is created to offload PA_VALIDATE rules to NPU. DashOffloadOrch listens to DASH_PA_VALIDATION_TABLE entry updates in APPL_STATE_DB. Orchagent creates a new ACL table "PA_VALIDATION" and bind to all DPU connected ports in the EGRESS direction. For every Source IP address in the DASH_PA_VALIDATION rule entry, orchagent creates ACL rules:
+## 2.2 PA Validation Offload
+
+A new orchagent (PaValidationOffloadOrch) is responsible for offloading the PA_VALIDATION table entries. Once started by DashOffloadManager, it subscribes to the PA_VALIDATION table and maps the PA validation entries into the ACL configuration.
+
+### 2.2.1 PA Validation Offload ACL configuration
+
+The offloading is achieved by using a User-Defined EGRESS ACL table bound to the NPU<->DPU port. The table is created/managed per DPU.
 
 ```
-for (SRC_IP : SRC_IP_LIST from the table)
-    - {VNI, SRC_IP} and PERMIT action with priority 100
-- {VNI} and DENY action with priority 10
+{
+    "ACL_TABLE_TYPE": {
+        "DASH_PA_VALIDATION": {
+            "MATCHES": [
+                "TUNNEL_VNI",
+                "SRC_IP",
+                "SRC_IPV6"
+            ],
+            "ACTIONS": [
+                "PACKET_ACTION"
+            ],
+            "BIND_POINTS": [
+                "PORT"
+            ]
+        }
+    }
+}
 ```
 
-Orchagent process Add/update/delete operations on the tables and update the ACL rules accordingly.
+Please refer to https://github.com/sonic-net/SONiC/blob/master/doc/acl/ACL-Table-Type-HLD.md for more details regarding User-Defined ACL tables.
 
-<img src="images/PA_Validate_DB_Orchagent.png">
+The ACL table configuration:
+
+```
+{
+  "ACL_TABLE": {
+    "DASH_PA_VALIDATION_DPU0": {
+      "STAGE": "EGRESS",
+      "TYPE": "DASH_PA_VALIDATION",
+      "PORTS": ["Ethernet0"]

```

Comment:
Is this binding port list based on DPU index or all the ports connected to the DPUs?

Reply:
To keep the config simple, can we bind to all DPU connected ports?

Reply:
Each DPU has its own table. The NPU<->DPU interface is taken from this configuration:
https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/pmon/smartswitch-pmon.md#313-npu-to-dpu-data-port-mapping

Having a single table and binding all the DPU-connected ports is actually more complicated. Consider the following:
1) DPU0 is configured with pa validation (vni 10, addr 1.1.1.1) (ACL rule is added to the table)
2) DPU1 is configured with the same pa validation entry (ACL table already has the rule)
3) DPU0 pa validation configuration is removed
Now, we cannot remove the ACL rule since it's needed for DPU1, and not removing it would result in incorrect behavior for DPU0.

It can be fixed with PORT_ID key though. I'm not sure if this approach has any benefit. Managing a table per DPU is straightforward, we have a zmq connection proxy per dpu, and offload orch in general per dpu.